mesos-rxjava-example/mesos-rxjava-example-framework/src/main/java/org/apache/mesos/rx/java/example/framework/sleepy/State.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.mesos.rx.java.example.framework.sleepy;
+
+import org.apache.mesos.v1.Protos;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newConcurrentMap;
+
+final class State {
+    private static final Logger LOGGER = LoggerFactory.getLogger(State.class);
+
+    private Protos.FrameworkID fwId;
+
+    @NotNull
+    private final Map<Protos.TaskID, Protos.TaskState> taskStates;
+
+    public State() {
+        this.taskStates = newConcurrentMap();
+    }
+
+    public Protos.FrameworkID getFwId() {
```

Comment:
Technically, you need to make this method `synchronized` to ensure that a reading thread sees the update from a writing thread. Or you could remove `synchronized` from `setFwId()` and make `fwId` an `AtomicReference` or `volatile`.